src/main/java/software/amazon/encryption/s3/legacy/internal/RangedGetUtils.java

```diff
@@ -18,13 +18,13 @@ public static long[] getRange(String range) {
         if (range == null) {
             return null;
         }
-        if (!range.matches("bytes=[0-9]+-[0-9]+")) {
+        if (!range.matches("^bytes=(\\d+-\\d+|\\d+-)$")) {
```

Comment:
nit: since this is a public method, we should add some unit tests around the range input and how it is handled

Reply:
agreed, will add some unit test cases around desired range and crypto range